pkg/cluster/database.go

```diff
@@ -101,14 +101,19 @@ func (c *Cluster) databaseAccessDisabled() bool {
 }
 
 func (c *Cluster) initDbConn() error {
+	if c.pgDb != nil {
+		return nil
+	}
+
 	return c.initDbConnWithName("")
 }
 
+// Worker function for connection initialization. This function does not check
+// if the connection is already open, if it is then it will be overwritten.
+// Callers need to make sure no connection is open, otherwise we could leak
+// connections
 func (c *Cluster) initDbConnWithName(dbname string) error {
 	c.setProcessName("initializing db connection")
-	if c.pgDb != nil {
```

Comment:
We can for now assume we do not have any kind of parallel work per Postgres cluster.

Thus we should for now close the connection and set to nil, if we end up here and that is not the case already. We have now fallen once over this code, not doing what it is supposed to do and pgDb being shared.

This is also no real overhead, since this code now always establishes a new connection.